### PR TITLE
Add metrics and chain package

### PR DESCRIPTION
**Pull request by wanliqun**
also:
1. refactor alert package to use config struct;
2. add support for package init with config injection for more flexibility.

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/go-conflux-util/4)

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
+package metrics
+
+import (
+	"time"
+
+	viperutil "github.com/Conflux-Chain/go-conflux-util/viper"
+	"github.com/ethereum/go-ethereum/metrics"
```
Comment (boqiu) [inline metrics/metrics.go:7]: Strongly recommend to use `github.com/rcrowley/go-metrics` instead, and remove helper methods in the bottom.

**Review by boqiu — CHANGES_REQUESTED** [review]

File: alert/alert.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package alert
+
+import (
+	viperutil "github.com/Conflux-Chain/go-conflux-util/viper"
+)
+
+// AlertConfig alert configuration such as DingTalk settings.,
```
[the remark this answers is not in the document]
Comment (wanliqun) [reply alert/alert.go:7]: ok

File: alert/alert.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	Enabled bool `default:"false" mapstructure:"enabled"`
+	// mobiles for @ members
+	AtMobiles []string `mapstructure:"atMobiles"`
+	// whether to @ all members
+	IsAtAll bool `default:"false" mapstructure:"isAtAll"`
+	// webhook url
+	Webhook string `mapstructure:"webhook"`
+	// secret token
+	Secret string `mapstructure:"secret"`
+}
```
[the remark this answers is not in the document]
Comment (wanliqun) [reply alert/alert.go:28]: ok

File: alert/alert.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	viperutil "github.com/Conflux-Chain/go-conflux-util/viper"
+)
+
+// AlertConfig alert configuration such as DingTalk settings.,
+type AlertConfig struct {
+	// Custom tags are usually used to differentiate between different networks and enviroments
+	// such as mainnet/testnet, prod/test/dev or any custom info for more details.
+	CustomTags []string `default:"[dev]" mapstructure:"customTags"`
```
[the remark this answers is not in the document]
Comment (wanliqun) [reply alert/alert.go:11]: done

File: alert/alert.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	// DingTalk settings
+	DingTalk *DingTalkConfig `mapstructure:"dingtalk"`
```
[the remark this answers is not in the document]
Comment (wanliqun) [reply alert/alert.go:13]: Extracting `DingTalkConfig` from AlertConfig is good for separation of concern and more extensible.

File: chain/client.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
+package chain
+
+import (
+	"time"
+
+	sdk "github.com/Conflux-Chain/go-conflux-sdk"
+	viperutil "github.com/Conflux-Chain/go-conflux-util/viper"
```
[the remark this answers is not in the document]
Comment (wanliqun) [reply chain/client.go:7]: done

File: chain/client.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	err := viperutil.UnmarshalKey("cfx", &option)
+	if err != nil {
+		return sdk.ClientOption{}, errors.WithMessage(err, "failed to unmarshal Conflux client config")
+	}
+
+	return sdkClientOptionFromConfig(&option), nil
+}
+
+// sdkClientOptionFromConfig generates sdk client option from config.
+func sdkClientOptionFromConfig(config *CfxClientConfig) sdk.ClientOption {
```
[the remark this answers is not in the document]
Comment (wanliqun) [reply chain/client.go:66]: done

File: metrics/metrics.go (hunk 1 above)
Comment (wanliqun) [reply metrics/metrics.go:7]: `github.com/rcrowley/go-metrics` seems not to be rather robust with influxdb integration.

File: metrics/metrics.go [hunk 8]
```diff
@@ -25,25 +25,25 @@ var (
 // MetricsConfig metrics configurations such as influxdb settings.,
 type MetricsConfig struct {
 	// switch to turn on or off metrics
-	Enabled bool `default:"false" mapstructure:"enabled"`
+	Enabled bool
 	// namespace for metrics reporting
-	Namespace string `default:"" mapstructure:"namespace"`
+	Namespace string `default:"cfx"`
```
Comment (boqiu) [inline metrics/metrics.go:30]: Better to use empty as namespace.

**Review by boqiu — CHANGES_REQUESTED** [review]

File: metrics/metrics.go (hunk 8 above)
Comment (wanliqun) [reply metrics/metrics.go]: done

**Review by boqiu — APPROVED** [review]

**Pull request merged**